automated install/basic-install.sh

```diff
@@ -113,2765 +113,2939 @@ runUnattended=false
 INSTALL_WEB_SERVER=true
 # Check arguments for the undocumented flags
 for var in "$@"; do
-    case "$var" in
-        "--reconfigure" ) reconfigure=true;;
-        "--unattended" ) runUnattended=true;;
-        "--disable-install-webserver" ) INSTALL_WEB_SERVER=false;;
-    esac
+  case "$var" in
+    "--reconfigure" ) reconfigure=true;;
+    "--unattended" ) runUnattended=true;;
+    "--disable-install-webserver" ) INSTALL_WEB_SERVER=false;;
+  esac
 done
 
 # If the color table file exists,
 if [[ -f "${coltable}" ]]; then
-    # source it
-    source "${coltable}"
+  # source it
+  source "${coltable}"
 # Otherwise,
 else
-    # Set these values so the installer can still run in color
-    COL_NC='\e[0m' # No Color
-    COL_LIGHT_GREEN='\e[1;32m'
-    COL_LIGHT_RED='\e[1;31m'
-    TICK="[${COL_LIGHT_GREEN}✓${COL_NC}]"
-    CROSS="[${COL_LIGHT_RED}✗${COL_NC}]"
-    INFO="[i]"
-    # shellcheck disable=SC2034
-    DONE="${COL_LIGHT_GREEN} done!${COL_NC}"
-    OVER="\\r\\033[K"
+  # Set these values so the installer can still run in color
+  COL_NC='\e[0m' # No Color
+  COL_LIGHT_GREEN='\e[1;32m'
+  COL_LIGHT_RED='\e[1;31m'
+  TICK="[${COL_LIGHT_GREEN}✓${COL_NC}]"
+  CROSS="[${COL_LIGHT_RED}✗${COL_NC}]"
+  INFO="[i]"
+  # shellcheck disable=SC2034
+  DONE="${COL_LIGHT_GREEN} done!${COL_NC}"
+  OVER="\\r\\033[K"
+fi
+
+# Check the init system
+if is_command systemctl; then
```

Comment:
In the tests this is appearing:

```
", stderr=b'/opt/pihole/basic-install.sh: line 142: is_command: command not found
/opt/pihole/basic-install.sh: line 144: is_command: command not found
```

Because you are trying to use the command `is_command` before it is declared (further on down the page)